os/arch/arm/src/armv7-r/arm_assert.c

```diff
@@ -208,112 +211,68 @@ static int is_text_address(unsigned long programCounter)
  * Below API works if there is existance of System.map file in rom fs
  ****************************************************************************/
 #ifdef CONFIG_DEBUG_DISPLAY_SYMBOL
-int get_symbol(unsigned long search_addr, char *buffer, size_t buflen)
+void up_symtab_init(void)
 {
-	FILE *pFile;
-	int total;
-	unsigned long sym_offset;
-	unsigned long total_size;
-	unsigned long addr;
-	unsigned long next_addr;
-	int first = 0;
-	int last;
-	int mid;
-	char line[128] = { '\0' };
-	char data[6][128] = { {'\0', '\0'} };
-	char c;
-	int word;
-	int ch_in_word;
-	int read_line;
-	int ch_in_line;
-
-	if (recursive_abort) {
-		/* If there is a crash in file system, below operation would
-		   lead to recursive abort, as this api will be called inside
-		   abort handler and tries to access the file system api.
-		   To avoid recursive abort, just return from here if we are
-		   already in abort mode.
-		 */
-		return -1;
+	if (up_symtab_flashdrv_init() == OK) {
+		symtab_init = true;
 	}
+}
 
-	pFile = fopen("/rom/System.map", "r");
+static unsigned long getaddr(int index)
+{
+	unsigned long addr;
+	up_symtab_flashdrv_read(&addr, (index - 1) * 4 + 4, 4);
+	return addr;
+}
 
-	/* Check if file exists */
-	if (pFile == NULL) {
-		lldbg("Could not open file: /rom/System.map\n");
-		return -1;
-	}
-	// obtain file size:
-	fseek(pFile, 0, SEEK_END);
-	last = ftell(pFile);
+static void getname(int ind, char *buffer, size_t buflen)
+{
+	unsigned long sym_offlen;
+	unsigned long nextsym_offlen;
+	unsigned long symlen;
+
```

Comment:
i feel input arguments NULL checking should be performed before use
What your opinion ?

Reply:
Yes, input parameters need to checked.